crates/bevy_app/src/terminal_ctrl_c_handler.rs

```diff
@@ -0,0 +1,63 @@
+use std::sync::atomic::{AtomicBool, Ordering};
+
+use bevy_ecs::event::EventWriter;
+
+use crate::{App, AppExit, Plugin, Update};
+
+#[cfg(not(target_arch = "wasm32"))]
+/// Indicates that all [`App`]'s should exit.
+static RECEIVED_CTRL_C: AtomicBool = AtomicBool::new(false);
+
+/// Gracefully handles `Ctrl+C` by emitting a [`AppExit`] event. This plugin is part of the `DefaultPlugins`.
+///
+/// ```no_run
+/// # use bevy_app::{App, NoopPluginGroup as MinimalPlugins, PluginGroup, TerminalCtrlCHandlerPlugin};
+/// fn main() {
+///     App::new()
+///         .add_plugins(MinimalPlugins)
+///         .add_plugins(TerminalCtrlCHandlerPlugin)
+///         .run();
+/// }
+/// ```
+///
+/// If you want to setup your own `Ctrl+C` handler, you should disable this
+/// plugin from `DefaultPlugins`:
+/// ```no_run
+/// # use bevy_app::{App, NoopPluginGroup as DefaultPlugins, PluginGroup, TerminalCtrlCHandlerPlugin};
+/// fn main() {
+///     App::new()
+///         .add_plugins(DefaultPlugins.build().disable::<TerminalCtrlCHandlerPlugin>())
+///         .run();
+/// }
+/// ```
+#[derive(Default)]
+pub struct TerminalCtrlCHandlerPlugin;
+
+#[cfg(not(target_arch = "wasm32"))]
+impl TerminalCtrlCHandlerPlugin {
+    /// Handler for when the user presses `Ctrl+C` on the terminal. Should be called from the `Ctrl+C` hook.
+    pub fn on_ctrl_c() {
```

Comment:
Is there a reason this is public? I'd rather hide it if not.

Reply:
See their migration guide. If I understand it correctly, the idea is that you can still call this manually to notify all apps even when you're doing your own, manual CTRL-C handling.

Reply:
I renamed it but the idea was that if a user defines their own ctrlc handler then they will be able to use the plugins exit function to in that handler to exit bevy. They would not have to setup their own system and static flag.